homeassistant/components/solaredge/coordinator.py

```diff
@@ -143,9 +139,13 @@ def update(self) -> None:
             ]:
                 self.attributes[key] = value
             elif key == "status":
-                self.data = value
+                self.data["status"] = value
```

Comment:
I have adjust SolarEdgeDetailsDataService to return a `dict` like all other coordinators.
This avoids adding conditions all around for handling `str | dict | None`